### ISSUE TITLE
Inconsistency in trace structure 

### ISSUE THREAD

**Issue by maximzubkov** [vespa-engine/vespa#33017]
Hello, Vespa Team and thanks a lot for your work! 

With the recent changes to the Vespa trace structure introduced in this PR https://github.com/vespa-engine/vespa/pull/32914, I've noticed that the format of different traces has become slightly inconsistent. Specifically, the changes include:
```diff
{
    "start_time": "2024-12-09 15:04:15.572 UTC",
    "traces": [
+        {
+            "timestamp_ms": 0.503333,
+            "event": "searching for 10 hits at offset 0 (with grouping)"
+        },
        {
            "traces": [...],
            "timestamp_ms": 2.918041,
            "tag": "query_setup"
        },
        {
            "timestamp_ms": 2.949916,
            "tag": "query_execution_plan",
            "optimized": {...}
        },
        {
            "timestamp_ms": 4.589458,
            "tag": "query_execution",
            "threads": [...]
        },
+        {
+            "timestamp_ms": 4.705125,
+            "event": "returning 10 hits from total 214"
+        }
    ],
    "distribution-key": 0,
    "document-type": "index_ebb8d0ee_73d7_41c6_9c98_adb81429c786",
    "duration_ms": 4.706875
}
```
Would it make sense to introduce a `tag` field in the first and last traces for consistency? Thank you, and looking forward to your response

**Comment by geirst** [comment]
@havardpe and @arnej27959 : Do you have a suggestion of tags we could use for the new trace events?

**Comment by arnej27959** [comment]
Note that there are many events in the trace and they won't have a "tag", so that part was consistent already.  But I guess adding tags for these events can be useful, so we'll do that.

**Comment by maximzubkov** [comment]
Thank you!

**Comment by geirst** [comment]
Fixed in Vespa 8.459.3.
